lib/full-site-editing/edit-site-page.php

```diff
@@ -263,3 +263,18 @@ function register_site_editor_homepage_settings() {
 	);
 }
 add_action( 'init', 'register_site_editor_homepage_settings', 10 );
+
+/**
+ * Sets the HTML <title> in the Site Editor list page to be the title of the CPT
+ * being edited, e.g. 'Templates'.
+ */
+function gutenberg_set_site_editor_list_page_title() {
+	global $title;
+	if ( gutenberg_is_edit_site_list_page() ) {
+		$post_type = get_post_type_object( $_GET['postType'] );
+		if ( $post_type ) {
+			$title = $post_type->labels->name;
+		}
+	}
+}
+add_action( 'load-appearance_page_gutenberg-edit-site', 'gutenberg_set_site_editor_list_page_title' );
```

Comment:
Any reason for not using the `admin_title` filter - https://developer.wordpress.org/reference/hooks/admin_title/?

Reply:
I wanted to write this in a way that's pretty close to how it will look when added to Core, which is that we'll just change `$title` here:

https://github.com/WordPress/wordpress-develop/blob/37abb3a4707ea81696eea516b84e0cfc7c70327d/src/wp-admin/site-editor.php#L27